src/arrayMethodJoin.js

```diff
@@ -4,8 +4,20 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] !== null && this[i] !== undefined) {
+        result += this[i];
+      }
+
+      if (this.length > 1 & i < this.length - 1) {
```

Comment:
redundant check, if length is 1, then i < this.length will alway be false as i can be equal to 0 only, and 0 < 1 - 1 is false
```suggestion
      if (i < this.length - 1) {
```